File: langgraph/data/equiv/7.py
from typing import List


def extract_matching_strings(sequence: List[str], pattern: str) -> List[str]:
    result = []
    index = 0
    while index < len(sequence):
        if pattern in sequence[index]:
            result.append(sequence[index])
        index += 1
    return result
# method name: filter_by_substring -> extract_matching_strings
# parameter name: strings -> sequence, substring -> pattern
# changed list comprehension to while loop for iteration
# introduced an explicit result list to append matching strings

File: langgraph/result/qwen/32b-eq2js-10/7.js
function extractMatchingStrings(sequence, pattern) {
    const result = [];
    let index = 0;
    while (index < sequence.length) {
        if (sequence[index].includes(pattern)) {
            result.push(sequence[index]);
        }
        index += 1;
    }
    return result;
}